op-challenger/game/fault/agent.go

```diff
@@ -126,6 +133,41 @@ func (a *Agent) tryResolve(ctx context.Context) bool {
 	return true
 }
 
+func (a *Agent) resolveClaims(ctx context.Context) error {
+	claims, err := a.loader.FetchClaims(ctx)
+	if err != nil {
+		return fmt.Errorf("failed to fetch claims: %w", err)
+	}
+	if len(claims) == 0 {
+		return nil
+	}
+
+	var resolvableClaims []int64
+	for _, claim := range claims {
+		a.log.Debug("checking if claim is resolvable", "claimIdx", claim.ContractIndex)
+		if err := a.responder.CallResolveClaim(ctx, int64(claim.ContractIndex)); err == nil {
+			a.log.Info("Resolving claim", "claimIdx", claim.ContractIndex)
+			resolvableClaims = append(resolvableClaims, int64(claim.ContractIndex))
+		}
+	}
+	a.log.Info("Resolving claims", "numClaims", len(resolvableClaims))
+
+	var wg sync.WaitGroup
```

Comment:
Resolving claims is done asynchronously to speed up resolution. Otherwise, in large games containing N claims, it'll take at least N blocks for the entire game to be resolved.
Instead, this lets us resolve the game resolving a game in roughly N blocks, where N is the max game depth.

Reply:
We should be able to batch these actions in a single multicall3 call if we order them correctly

Reply:
But that improvement can be ticketed and done later

Reply:
Yeah this can be improved. Tho we probably shouldn't use multicall3 due to https://github.com/ethereum-optimism/client-pod/issues/61. Given multiple players, it's likely that a single call fails - which short-circuits the entire transaction.